ParU fails with OpenMP from MSVC

**Describe the bug**
When trying to build ParU with an MSVC toolchain, compilation fails with errors like the following:
```
  D:\a\SuiteSparse\SuiteSparse\ParU\Source\paru_analyze.cpp(1060): error C3001: 'taskloop': expected an OpenMP directive name
  D:\a\SuiteSparse\SuiteSparse\ParU\Source\paru_analyze.cpp(1068): error C3001: 'taskloop': expected an OpenMP directive name
  D:\a\SuiteSparse\SuiteSparse\ParU\Source\paru_analyze.cpp(1300): error C3001: 'taskloop': expected an OpenMP directive name
  D:\a\SuiteSparse\SuiteSparse\ParU\Source\paru_analyze.cpp(1314): error C3001: 'taskloop': expected an OpenMP directive name
  [3/84] Building CXX object CMakeFiles\paru.dir\Release\Source\paru_assemble.cpp.obj
  FAILED: CMakeFiles/paru.dir/Release/Source/paru_assemble.cpp.obj 
```
Afaict, the `taskloop` construct was introduced with OpenMP 4.5. MSVC supports only OpenMP 2.0.

**To Reproduce**
Try to build ParU with OpenMP using MSVC (see CI) or another toolchain that lacks support for OpenMP 4.5.

**Expected behavior**
ParU builds without error.

**Desktop (please complete the following information):**
 - OS: Windows
 - compiler MSVC 2022

Thanks for clarifying. I opened #463 that should fix this.

It looks like building LAGraph with OpenMP also doesn't work.
Error:
```
  [12/372] Building C object src\CMakeFiles\lagraph.dir\Release\algorithm\LG_CC_FastSV6.c.obj
  FAILED: src/CMakeFiles/lagraph.dir/Release/algorithm/LG_CC_FastSV6.c.obj 
  C:\Miniconda\envs\test\Library\bin\ccache.exe C:\PROGRA~1\MICROS~2\2022\ENTERP~1\VC\Tools\MSVC\1435~1.322\bin\HostX64\x64\cl.exe  /nologo -D_CRT_SECURE_NO_WARNINGS -Dlagraph_EXPORTS -DCMAKE_INTDIR=\"Release\" -ID:\a\SuiteSparse\SuiteSparse\LAGraph\build\src -ID:\a\SuiteSparse\SuiteSparse\LAGraph\src -ID:\a\SuiteSparse\SuiteSparse\LAGraph\include -ID:\a\SuiteSparse\SuiteSparse\LAGraph\src\utility -ID:\a\SuiteSparse\SuiteSparse\LAGraph\test\include -ID:\a\SuiteSparse\SuiteSparse\GraphBLAS\Include -I\usr\local\include -ID:\a\SuiteSparse\SuiteSparse\LAGraph\src\include /DWIN32 /D_WINDOWS /W3 -DLGDIR=D:/a/SuiteSparse/SuiteSparse/LAGraph /MD /O2 /Ob2 /DNDEBUG /MD /O2 /Ob2 /DNDEBUG -std:c11 -openmp /showIncludes /Fosrc\CMakeFiles\lagraph.dir\Release\algorithm\LG_CC_FastSV6.c.obj /Fdsrc\CMakeFiles\lagraph.dir\Release\ /FS -c D:\a\SuiteSparse\SuiteSparse\LAGraph\src\algorithm\LG_CC_FastSV6.c
  D:\a\SuiteSparse\SuiteSparse\LAGraph\src\utility\LG_internal.h(390): warning C4244: '=': conversion from 'double' to 'int64_t', possible loss of data
  D:\a\SuiteSparse\SuiteSparse\LAGraph\src\algorithm\LG_CC_FastSV6.c(297): warning C4244: '=': conversion from 'GrB_Index' to 'int', possible loss of data
  D:\a\SuiteSparse\SuiteSparse\LAGraph\src\algorithm\LG_CC_FastSV6.c(421): error C3015: initialization in OpenMP 'for' statement has improper form
  D:\a\SuiteSparse\SuiteSparse\LAGraph\src\algorithm\LG_CC_FastSV6.c(446): error C3015: initialization in OpenMP 'for' statement has improper form
  D:\a\SuiteSparse\SuiteSparse\LAGraph\src\algorithm\LG_CC_FastSV6.c(556): error C3015: initialization in OpenMP 'for' statement has improper form
  D:\a\SuiteSparse\SuiteSparse\LAGraph\src\algorithm\LG_CC_FastSV6.c(611): error C3015: initialization in OpenMP 'for' statement has improper form
```

Not sure (yet) what that means. "improper form".
A version issue again?

--- a/ParU/CMakeLists.txt
+++ b/ParU/CMakeLists.txt
@@ -54,10 +54,21 @@ endif ( )
 option ( NOPENMP "ON: do not use OpenMP.  OFF (default): use OpenMP" OFF )
 if ( NOPENMP )
     # OpenMP has been disabled
-    set ( OPENMP_CXX_FOUND OFF )
-    set ( OPENMP_C_FOUND OFF )
+    set ( OpenMP_CXX_FOUND OFF )
+    set ( OpenMP_C_FOUND OFF )
 else ( )
     find_package ( OpenMP )
+    if ( OpenMP_CXX_FOUND AND OpenMP_CXX_VERSION VERSION_LESS 4.5 )
+        set ( NOPENMP ON )
+    endif ( )
+    if ( DEMO AND OpenMP_C_FOUND AND OpenMP_C_VERSION VERSION_LESS 4.5 )
+        set ( NOPENMP ON )
+    endif ( )
+    if ( NOPENMP )
+        message ( STATUS "Version of OpenMP is too old." )
+        set ( OpenMP_CXX_FOUND OFF )
+        set ( OpenMP_C_FOUND OFF )
+    endif ( )
 endif ( )
 
 find_package ( SuiteSparse_config 7.3.0
@@ -157,7 +168,7 @@ if ( NOT NSTATIC )
 endif ( )
 
 # OpenMP:
-if ( OPENMP_CXX_FOUND )
+if ( OpenMP_CXX_FOUND )
     message ( STATUS "OpenMP C++ libraries:    ${OpenMP_CXX_LIBRARIES} ")
     message ( STATUS "OpenMP C++ include:      ${OpenMP_CXX_INCLUDE_DIRS} ")
     message ( STATUS "OpenMP C++ flags:        ${OpenMP_CXX_FLAGS} ")
@@ -243,13 +254,13 @@ if ( DEMO )
     # Demo programs
     #---------------------------------------------------------------------------
 
-    if ( OPENMP_CXX_FOUND )
+    if ( OpenMP_CXX_FOUND )
         add_executable ( paru_demo    "Demo/paru_demo.cpp"   )
         target_link_libraries ( paru_demo
             PUBLIC paru SuiteSparse::CHOLMOD SuiteSparse::UMFPACK OpenMP::OpenMP_CXX )
     endif ( )
 
-    if ( OPENMP_C_FOUND )
+    if ( OpenMP_C_FOUND )
         add_executable ( paru_democ   "Demo/paru_democ.c"    )
         target_link_libraries ( paru_democ
             PUBLIC paru SuiteSparse::CHOLMOD SuiteSparse::UMFPACK OpenMP::OpenMP_C )

